[visionOS] When transitioning from video fullscreen to element fullscreen, the element fullscreen window is not frontmost

aestes opened the pull request:
#### 4fb78297f66382824d32daea416bcdccc940a801
<pre>
[visionOS] When transitioning from video fullscreen to element fullscreen, the element fullscreen window is not frontmost
<a href="https://bugs.webkit.org/show_bug.cgi?id=273100">https://bugs.webkit.org/show_bug.cgi?id=273100</a>
<a href="https://rdar.apple.com/126642251">rdar://126642251</a>

Reviewed by Wenson Hsieh and Aditya Keerthi.

When entering video fullscreen LinearMediaKit changes the client scene&apos;s windows&apos; transform3D, then
fails to restore them to their original values when exiting fullscreen. As a result, the element
fullscreen window may no longer be frontmost when exiting back to element fullscreen.

This issue was originally worked around in 277031@main by exiting element fullscreen when entering
video fullscreen, but this was suboptimal since users expect to return to element fullscreen if
that&apos;s the state they were in when entering video fullscreen. This change improves the workaround by
instead restoring the expected transform3Ds on both the element fullscreen window and WKWebView&apos;s
window when exiting video fullscreen.

Note that this change also removes dead code in -[WKContentViewInteraction _did(Enter|Exit)Fullscreen]
so that hooking up PageClientImpl::did(Enter|Exit)Fullscreen does not cause an unintended behavior
change.

* Source/WebKit/UIProcess/Cocoa/PlaybackSessionManagerProxy.mm:
(WebKit::PlaybackSessionManagerProxy::setVideoReceiverEndpoint):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::didEnterFullscreen):
(WebKit::WebPageProxy::didExitFullscreen):
* Source/WebKit/UIProcess/ios/PageClientImplIOS.h:
* Source/WebKit/UIProcess/ios/PageClientImplIOS.mm:
(WebKit::PageClientImpl::didExitFullscreen):
(WebKit::PageClientImpl::didEnterFullscreen): Deleted.
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _didEnterFullscreen]): Deleted.
(-[WKContentView _didExitFullscreen]): Deleted.
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.h:
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(-[WKFullScreenWindowController didExitFullscreen]):

Canonical link: <a href="https://commits.webkit.org/277877@main">https://commits.webkit.org/277877@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/be70f9cf38df3a351f3b10327d38c43aab68b34c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48809 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28021 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/51775 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51495 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44875 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/33964 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25550 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/39898 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/49391 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/25665 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42085 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20999 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/23134 "Passed tests") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6864 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/45062 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/43755 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/53411 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/23860 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20129 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/47202 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/25123 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/42292 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/46147 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10755 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/25930 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24843 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh reviewed (approved):
r=mews!

It looks like we might need some `ENABLE(VIDEO_PRESENTATION_MODE)` guards.

aestes committed:
[visionOS] When transitioning from video fullscreen to element fullscreen, the element fullscreen window is not frontmost
https://bugs.webkit.org/show_bug.cgi?id=273100
rdar://126642251

Reviewed by Wenson Hsieh and Aditya Keerthi.

When entering video fullscreen LinearMediaKit changes the client scene's windows' transform3D, then
fails to restore them to their original values when exiting fullscreen. As a result, the element
fullscreen window may no longer be frontmost when exiting back to element fullscreen.

This issue was originally worked around in 277031@main by exiting element fullscreen when entering
video fullscreen, but this was suboptimal since users expect to return to element fullscreen if
that's the state they were in when entering video fullscreen. This change improves the workaround by
instead restoring the expected transform3Ds on both the element fullscreen window and WKWebView's
window when exiting video fullscreen.

Note that this change also removes dead code in -[WKContentViewInteraction _did(Enter|Exit)Fullscreen]
so that hooking up PageClientImpl::did(Enter|Exit)Fullscreen does not cause an unintended behavior
change.

* Source/WebKit/UIProcess/Cocoa/PlaybackSessionManagerProxy.mm:
(WebKit::PlaybackSessionManagerProxy::setVideoReceiverEndpoint):
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::didEnterFullscreen):
(WebKit::WebPageProxy::didExitFullscreen):
* Source/WebKit/UIProcess/ios/PageClientImplIOS.h:
* Source/WebKit/UIProcess/ios/PageClientImplIOS.mm:
(WebKit::PageClientImpl::didExitFullscreen):
(WebKit::PageClientImpl::didEnterFullscreen): Deleted.
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _didEnterFullscreen]): Deleted.
(-[WKContentView _didExitFullscreen]): Deleted.
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.h:
* Source/WebKit/UIProcess/ios/fullscreen/WKFullScreenWindowControllerIOS.mm:
(-[WKFullScreenWindowController didExitFullscreen]):

Canonical link: https://commits.webkit.org/277877@main
